src/vehicle.cpp

```diff
@@ -3572,6 +3618,63 @@ void vehicle::power_parts()
         }
     }
 
+    if( has_part( "BOILER", true ) ) {
+        for( size_t b=0 ; b < boilers.size(); ++b ) {
+            printf("\nchecking boiler %zu\n", b);
+            if( !is_boiler_on( b ) ) {
+                printf("boiler is off\n");
+                continue;
+            }
+
+            printf("looking for tank\n");
+            const vehicle_part &pt = parts[boilers[b]];
+            auto tank = std::find_if( parts.begin(), parts.end(),
+                      [&pt](const vehicle_part &e){
+                          return pt.mount == e.mount && e.is_tank() && \
+                          e.can_reload("steam");
+                      });
+            if( tank == parts.end() ) {
+                printf("skipping boiler due to lack of tank space\n");
+                continue;
+            }
+
+            //calculate tank space avaliable
+            const float space_left = tank->ammo_capacity() - tank->ammo_remaining();
+
+            if(space_left < 1700){
+                printf("skipping boiler due to lack of tank space @ %i of %i\n",
+                       (int)tank->ammo_remaining(), (int)tank->ammo_capacity());
+                continue;
+            }
+
+            printf("tank space avaliable: %f\n", space_left);
+
+            //now consume fuel to create steam
+            const float multiplier = 1700;
+            const vpart_info &vpi = pt.info();
+            float energy_request = std::min<float>(space_left / multiplier, vpi.power);
```

Comment:
Don't use explicit template arguments for `std::min` and similar. Instead cast the two arguments to the same type (in this case: cast the non-float value to float). If someone later changes the type of `vpi.power` from `int` to `double`, the explicit template argument will convert it back to `float` (thereby reducing accuracy).